jwst/assign_wcs/assign_wcs_step.py

```diff
@@ -63,10 +63,16 @@ def process(self, input):
                 reference_file_names[reftype] = reffile if reffile else ""
 
             # Get the MSA metadata file if needed and add to reffiles
-            msa_metadata_file = input_model.meta.instrument.msa_metadata_file
-            if msa_metadata_file is not None:
-                msa_metadata_file = self.make_input_path(msa_metadata_file)
-                reference_file_names['msametafile'] = msa_metadata_file
+            if input_model.meta.exposure.type == "NRS_MSASPEC":
+                msa_metadata_file = input_model.meta.instrument.msa_metadata_file
+                if msa_metadata_file is not None and msa_metadata_file.strip() not in ["", "N/A"]:
+                    msa_metadata_file = self.make_input_path(msa_metadata_file)
+                    reference_file_names['msametafile'] = msa_metadata_file
+                else:
+                    log.info("MSA metadata file (MSAMETFL) is required for NRS_MSASPEC exposures.")
```

Comment:
Might want to bump this up to warning level, if not error.

Reply:
done - it's log.error now